crates/pop-parachains/src/call/mod.rs

```diff
@@ -46,6 +46,36 @@ pub fn construct_sudo_extrinsic(xt: DynamicPayload) -> DynamicPayload {
 	subxt::dynamic::tx("Sudo", "sudo", [xt.into_value()].to_vec())
 }
 
+/// Constructs a Proxy call extrinsic.
+///
+/// # Arguments
+/// * `pallets`: List of pallets available within the chain's runtime.
+/// * `proxied_account` - The account on whose behalf the proxy will act.
+/// * `xt`: The extrinsic representing the dispatchable function call to be dispatched using the
+///   proxy.
+pub fn construct_proxy_extrinsic(
+	pallets: &[Pallet],
+	proxied_account: String,
+	xt: DynamicPayload,
+) -> Result<DynamicPayload, Error> {
+	let proxy_function = find_dispatchable_by_name(pallets, "Proxy", "proxy")?;
+	// `find_dispatchable_by_name` doesn't support parsing parameters that are calls.
+	// Therefore, we only parse the first two parameters for the proxy call
+	// using `parse_dispatchable_arguments`, while the last parameter (which is the call)
+	// must be manually added.
+	let required_params: Vec<Param> = proxy_function.params.iter().take(2).cloned().collect();
```

Comment:
Can you not consume the params collection with into_iter() to avoid cloning?

Reply:
I can't do it because `find_dispatachable_by_name` returns a `&Function`